index.js

```diff
@@ -257,31 +255,62 @@ ShelfFirst.prototype.visibleStartingIndex = function(topOffset, width) {
     this.width = width;
   }
 
-  var top = 0;
-  var position, entry;
-  var previousTop = 0;
-  var index = -1;
+  var height = this.height();
   var length = this.length();
 
-  // TODO: implement fuzzy binary search for large N to reduce cost when search
-  // space is large. Should get this closer to O(log n)
+  // Start searching using the last item in the list
+  // and the bottom of the list for calculating the average hegiht
```

Comment:
should likely leave the `O(log n)` goal (or something more accurate) statement so future travelers understand the motivation for the bellow algorithim